### PR TITLE
SPS Week 2 Server

### PR DESCRIPTION
**Pull request by tianyuchen01**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/RandomMsgServlet.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  /**
+   * Convert an ArrayList of String to its json encoding.
+   * 
+   * @param list an ArrayList of String
+   * @return the json encoding String of the ArrayList of String
+   */
+  public String convertToJson(ArrayList<String> list) {
+    String json = "{";
```
Comment (bpytlik) [inline portfolio/src/main/java/com/google/sps/servlets/RandomMsgServlet.java]: It'd be better to use the gson library to convert the arrays to JSON than doing the conversion by hand.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -27,8 +27,45 @@ function addRandomGreeting() {
...
+    const randomIdx = Math.floor(Math.random() * 3);
+    if (randomIdx == 0) {
+        randomFactContainer.innerText = facts.learning;
+    }
+    else if (randomIdx == 1) {
+        randomFactContainer.innerText = facts.programmingLanguage;
+    }
+    else if (randomIdx == 2) {
+        randomFactContainer.innerText = facts.animal;
+    }
```
Comment (bpytlik) [inline portfolio/src/main/webapp/script.js]: If you use gson to do the conversion from the array to string on the server side, then you can use randomIdx to index directly into the array returned from the server.

**Review by bpytlik — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (tianyuchen01) [reply portfolio/src/main/webapp/script.js]: Wow! Thank you for telling me this usage! I didn't notice that parsing JSON could not only give you an object with fields, but can also just give you an array.

File: portfolio/src/main/java/com/google/sps/servlets/RandomMsgServlet.java (hunk 1 above)
Comment (tianyuchen01) [reply portfolio/src/main/java/com/google/sps/servlets/RandomMsgServlet.java]: Ok!!

**Pull request merged**